app/src/main/java/io/appium/uiautomator2/server/ServerConfig.java

```diff
@@ -16,10 +16,106 @@
 
 package io.appium.uiautomator2.server;
 
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.ConcurrentMap;
+
 public class ServerConfig {
-    private final static int PORT = 6790;
+    // server port (default: `6790`)
+    private final static int SERVER_PORT =
+        System.getenv("SERVER_PORT") == null ? 6790 :
+            Integer.parseInt(System.getenv("SERVER_PORT"));
+    // mjpeg server port (default: `7810`)
+    private final static int MJPEG_SERVER_PORT =
+        System.getenv("MJPEG_SERVER_PORT") == null ? 7810 :
+            Integer.parseInt(System.getenv("MJPEG_SERVER_PORT"));
+    // mjpeg stream will try to match this framerate (default: `10`)
+    private final static int MJPEG_SERVER_FRAMERATE =
+        System.getenv("MJPEG_SERVER_FRAMERATE") == null ? 10 :
+            Integer.parseInt(System.getenv("MJPEG_SERVER_FRAMERATE"));
+    // mjpeg scale should be between 1 and 100 (default: `50`)
+    private final static int MJPEG_SCALING_FACTOR =
```

Comment:
I would rather consider the actual screen density to calculate the default scale factor. By default uia takes sceenshots having `display.getRealSize(displaySize)` dimensions.

Reply:
To me this is configuration available to the integrator. While I agree basing it off the screen-density, etc. could be feasible I'd like to keep it as an absolute value.